Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -254,31 +325,72 @@ private final class MacroExpansionRewriter: SyntaxRewriter {
     // Swap in the argument for this parameter
     return arguments[parameterIndex].trimmed
   }
+
+  override func visit(_ node: GenericArgumentSyntax) -> GenericArgumentSyntax {
+    guard let parameterIndex = genericParameterReplacements[node] else {
+      return super.visit(node)
+    }
+
+    guard parameterIndex < genericArguments.count else {
+      return super.visit(node)
+    }
+
+    // Swap in the argument for type parameter
+    return GenericArgumentSyntax(
+      leadingTrivia: node.leadingTrivia,
+      node.unexpectedBeforeArgument,
+      argument: genericArguments[parameterIndex].trimmed,
+      node.unexpectedBetweenArgumentAndTrailingComma,
+      trailingComma: node.trailingComma,
+      node.unexpectedAfterTrailingComma
+        // TODO: seems we're getting spurious trailing " " here,
+        //  skipping trailing trivia for now
+        // trailingTrivia: node.trailingTrivia
+    )
```

Comment:
You should be able to simplify this slightly to

```suggestion
var node = node
    node.argument = genericArguments[parameterIndex].trimmed
    return node
```

Reply:
Ah, good call